### ISSUE TITLE
Consider moving to Group Normalization in the u-net due to usage of accumulative gradients.

### ISSUE THREAD

**Issue by vegovs** [gil-uav/semantic-image-segmentation#28]

**Comment by vegovs** [comment]
It's just that BN use more than just backpropagation information to update its parameters (the batch norm statistics).  So when you are using GA, you are not "accumulating" the BN statistics and so you don't solve the problem of using BN on a small batch size (it will still be like you are using a batch size of 4 basically). For the rest of the network, you are correct, it's like you are using a batch size of 16. With group norm, you don't have that problem, so if it's useful for you to have bigger batch size, than GN + GA will work out of the box if I'm not mistaken
